lib/config/configuration.go

```diff
@@ -2146,7 +2153,8 @@ func validateRoles(roles string) error {
 			defaults.RoleProxy,
 			defaults.RoleApp,
 			defaults.RoleDatabase,
-			defaults.RoleWindowsDesktop:
+			defaults.RoleWindowsDesktop,
+			defaults.RoleDiscovery:
```

Comment:
Is this necessary here? Would we ever expect a user to run `teleport start --roles=discovery` to run a standalone discovery service, or is discovery something that teleport should decide to enable based on the config?

Do we need to be able to generate join tokens for the discovery service, or does it always piggy back on another service?